source/server/listener_manager_impl.cc

```diff
@@ -380,11 +380,14 @@ void ListenerImpl::convertDestinationIPsMapToTrie() {
     for (const auto& entry : destination_ips_map) {
       std::vector<Network::Address::CidrRange> subnets;
       if (entry.first == EMPTY_STRING) {
-        list.push_back(
-            std::make_pair<ServerNamesMapSharedPtr, std::vector<Network::Address::CidrRange>>(
-                std::make_shared<ServerNamesMap>(entry.second),
-                {Network::Address::CidrRange::create("0.0.0.0/0"),
-                 Network::Address::CidrRange::create("::/0")}));
+        std::vector<Network::Address::CidrRange> cidr_ranges;
+        if (Network::Address::ipFamilySupported(AF_INET)) {
+          cidr_ranges.push_back(Network::Address::CidrRange::create("0.0.0.0/0"));
+        }
+        if (Network::Address::ipFamilySupported(AF_INET6)) {
+          cidr_ranges.push_back(Network::Address::CidrRange::create("::/0"));
+        }
+        list.push_back({std::make_shared<ServerNamesMap>(entry.second), cidr_ranges});
```

Comment:
Could you use the patch I provided (using unused `subnets` and sharing more of the code with the `!= EMPTY_STRING` path)? It's a bit more readable, IMHO.

Reply:
Updated. Please check